src/tests/reconnect.test.ts

```diff
@@ -0,0 +1,76 @@
+import FakeTimers from '@sinonjs/fake-timers'
+import test from 'ava'
+import Docker from 'dockerode'
+import { createClient } from 'redis'
+import transporter from '../index.js'
+
+const emit = () => undefined
+
+test.serial(
+  'should be able to reconnect to Redis if the server has disconnected',
+  async (t) => {
+    const clock = FakeTimers.install({ toFake: ['Date'], now: Date.now() })
+    // Create and connect the RedisClient
+    const redisClient = createClient()
+    await redisClient.connect()
+
+    const options = {
+      prefix: 'store',
+      redis: {
+        uri: 'redis://localhost:6380',
+      },
+      connectionTimeout: 1,
+    }
+
+    const pingAction = {
+      type: 'SERVICE',
+      payload: { type: 'ping' },
+      meta: {},
+    }
+    const connection = await transporter.connect(options, null, null, emit)
+    const firstActionResult = await transporter.send(pingAction, connection)
+
+    // Disconnect the Redis client from the network
+    const docker = new Docker({ socketPath: '/var/run/docker.sock' })
+    const networks: Docker.NetworkInspectInfo[] = await docker.listNetworks()
+    const network = networks.find(
+      (network) => network.Name === 'integreat-transporter-redis_reconnect',
+    )
+    t.truthy(network, 'Network not found')
+    const networkId = network?.Id ?? 'N/A'
+
+    await docker
+      .getNetwork(networkId)
+      .disconnect({ Container: 'reconnect-redis' })
+
+    // Wait for the connection to expire which should trigger an attempt to
+    // disconnect (and client.quit() or client.disconnect() under the hood)
+    // and then reconnect on the next call to transporter.connect()
+    clock.tick(options.connectionTimeout + 1)
```

Comment:
This can be replaced by:
`import { setTimeout as sleep } from 'timers/promises'` and
`await sleep(options.connectionTimeout + 1)` since it seems that it's really hard to get the FakeTimers to work

Reply:
Ok. :)